src/unicode/std_case_mapping_iter.rs

```diff
@@ -0,0 +1,136 @@
+// This file contains code extracted from std::char::CaseMappingIter.
```

Comment:
I hadn't checked through the need for this until just now, and I think I understand! Very cool.

So, to correct my understanding here, we need this since the iterator of a unicode titlecase because when we case map, we could end up with more unicode code points than what came in (e.g. 1 code point uppercase might be 3 lowercase chars). I'm trying to understand by reading the code through, but my brain is mush, and maybe it's easier for you to explain it to me - but my question is: Is it possible to wrap `core::char::ToLowerCase` and `core::char::ToUpperCase` iterators instead of bringing this in? It's entirely possible I'm missing something important here :sweat_smile:

Reply:
Title case is a distinct operation from upper and lowercase.

This character is the example I know:

> Dž (titlecase form; all-capitals form DŽ, lowercase dž)